outer.c

```diff
@@ -20,14 +20,27 @@
 #include <unistd.h>
 
 #include "capable.h"
+#include "compat.h"
 #include "enter.h"
 #include "outer.h"
 #include "path.h"
+#include "preprocessor.h"
 #include "sig.h"
 #include "userns.h"
 #include "util.h"
 #include "fd.h"
 
+# define REBIND_FDS_AND_CLOSE(Start, ...) \
+	do { \
+		int __fd_last = (Start); \
+		APPLY(_REBIND_FD, __VA_ARGS__) \
+		if (bst_close_range(__fd_last, ~0U, 0) == -1) { \
+			err(1, "close_range"); \
+		} \
+	} while (0)
+
+# define _REBIND_FD(Fd) Fd = dup2(Fd, __fd_last++);
```

Comment:
It feels kinda yucky that we want to mutate the caller's fd and for this reason we decide to do this in a macro. You could imagine all sorts of scenarios where this behavior becomes confusing


An alternate approach could be to just make it a regular function that takes in a list of FDs, then sorts, them, then calls bst_close_range on all the gaps in the sorted list. No need for macro madness (because the FDs don't need to change) and no need to mutate te inputs